### PR TITLE
Added gotchas for function variables

### LINKED ISSUE #16836: add gotcha for Function variables
```diff
diff --git a/doc/src/gotchas.rst b/doc/src/gotchas.rst
index 1d3052f..b2f6050 100644
--- a/doc/src/gotchas.rst
+++ b/doc/src/gotchas.rst
@@ -268,6 +268,33 @@ trick for getting tab completion in the regular Python console.
     See also `What is the best way to create symbols?
     <https://github.com/sympy/sympy/wiki/Faq>`_ in the FAQ.

+.. _symbols:
+
+Functions
+---------
+
+A function like ``f(x)`` can be created by defining the Function
+and the variable:
+
+    >>> from sympy import Function
+    >>> f = Function('f')
+    >>> x = Symbol('x')
+    >>> f(x)
+    f(x)
+
+If you assign ``f(x)`` to a Python variable `f` you will lose your
+ability to copy and paste that function or to create a function
+with a different argument: ``Function('f')`` is callable, but
+``Function('f')(x)`` is not:
+
+    >>> f = f(x)
+    >>> f
+    f(x)
+    >>> f(x)
+    Traceback (most recent call last):
+    ...
+    TypeError: 'f' object is not callable
+
 .. _symbolic-expressions:

 Symbolic Expressions
```

### PR DESCRIPTION
**Pull request by czgdp1807**
#### References to other Issues or PRs

Fixes https://github.com/sympy/sympy/issues/16836

#### Release Notes

NO ENTRY

### CODE REVIEW

File: doc/src/gotchas.rst [hunk 1]
```diff
@@ -270,6 +270,33 @@ trick for getting tab completion in the regular Python console.
...
+
+If you assign ``f(x)`` to a Python variable `f` you will lose your
+ability to copy and paste that function or to create a function
+with a different argument: ``Function('f')`` is callable, but
+``Function('f')(x)`` is not:
+
+    >>> f = f(x)
+    >>> f
+    f(x)
+    >>> f(x)
```
Comment (oscarbenjamin) [inline doc/src/gotchas.rst:295]: Perhaps a better demonstration would use a different argument e.g. `f(1)`.

Comment (czgdp1807) [reply doc/src/gotchas.rst:295]:
```suggestion
    >>> f(1)
```
Is the above okay?

Comment (oscarbenjamin) [reply doc/src/gotchas.rst:295]: Yes but the idea would be to show how it works in both cases.

Comment (czgdp1807) [reply doc/src/gotchas.rst:295]: The normal case is shown in L284 and L285. Should I add a normal call of `Function('f')(1)` here too?

Comment (oscarbenjamin) [reply doc/src/gotchas.rst:295]: The point is to show that if
```
f1 = Function('f')
f2 = Function('f')(x)
```
then you can call `f1` with different arguments e.g. `f1(1)` but that doesn't work for `f2(1)`. It is possible though with subs e.g. `f2.subs(x, 1)`.

Comment (czgdp1807) [reply doc/src/gotchas.rst]: I have changed the example.

File: doc/src/gotchas.rst [hunk 2]
```diff
@@ -270,6 +270,36 @@ trick for getting tab completion in the regular Python console.
...
+If you assign ``f(x)`` to a Python variable `f` you will lose your
+ability to copy and paste that function or to create a function
+with a different argument: ``Function('f')`` is callable, but
+``Function('f')(x)`` is not:
+
+    >>> f1 = Function('f1')
+    >>> f2 = Function('f2')('x')
+    >>> f1(1)
+    f1(1)
+    >>> f2(1)
```
Comment (oscarbenjamin) [inline doc/src/gotchas.rst:296]: It would also be good to show what happens when you output just `f1` and `f2`.

Comment (czgdp1807) [reply doc/src/gotchas.rst:300]: Added the outputs.

**Comment by czgdp1807** [comment]
Will merge tomorrow if no objections raised until then.

**Comment by oscarbenjamin** [comment]
Looks good.

**Pull request merged**